h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstPivot.java

```diff
@@ -0,0 +1,79 @@
+package water.rapids.ast.prims.mungers;
+
+import water.*;
+import water.fvec.*;
+import water.rapids.Rapids;
+import water.rapids.Env;
+import water.rapids.vals.ValFrame;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.util.VecUtils;
+
+public class AstPivot extends AstPrimitive {
+  @Override
+  public String[] args() {
+    return new String[]{"ary", "index", "column", "value"}; //the array and name of columns
+  }
+
+  @Override
+  public int nargs() {
+    return 1 + 4;
+  } // (pivot ary index column value)
+
+  @Override
+  public String str() {
+    return "pivot";
+  }
+
+  @Override
+  public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
+    Frame origfr = stk.track(asts[1].exec(env)).getFrame();
+    String _index = stk.track(asts[2].exec(env)).getStr();
+    String _column = stk.track(asts[3].exec(env)).getStr();
+    String _value = stk.track(asts[4].exec(env)).getStr();
+    Key k1 = Key.<Frame>make();
+    Frame fr = origfr.subframe(new String[]{_index,_column,_value}).deepCopy(k1.toString());
```

Comment:
I had to do this so that the Rapids.exec could see the Frame.  Im wondering if there is a better way.